internal/logger/logger.go

```diff
@@ -103,7 +103,14 @@ func (l *Logger) AddBlockStream(
 	if err != nil {
 		return err
 	}
-	defer f.Close()
+
+	defer func() {
+		err := f.Close()
+		if err != nil {
+			fmt.Println(err)
```

Comment:
nit: use `log.Fatal` or `log.Fatalln` here (it automatically calls `os.Exit(1)`). Here is an example https://github.com/coinbase/rosetta-cli/blob/4125bcbe7c60afc5296568721e019df4132f4f2e/cmd/check.go#L547